File: Week_02/n-ary-tree-postorder-traversal.java
    /*
    // Definition for a Node.
    class Node {
        public int val;
        public List<Node> children;

        public Node() {}

        public Node(int _val) {
            val = _val;
        }

        public Node(int _val, List<Node> _children) {
            val = _val;
            children = _children;
        }
    };
    */

    class Solution {
        public List<Integer> postorder(Node root) {
            List<Integer> lst =new ArrayList<Integer>();
            postorderP(root, lst );
            return lst;
        }
        public void postorderP(Node root, List<Integer> lst ) {
            if(null == root) 
            return;
            int i;
            for ( i = 0 ;  null != root.children && i < root.children.size(); i++) {
                postorderP(root.children.get(i), lst );
            }
            lst.add(root.val);
        }
    }

File: Week_02/n-ary-tree-postorder-traversal.cpp
/*
 *// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<int> postorder(Node* root) {
        vector<int> v;
        postorder(root, v) ;
        return  v;
    }
   void  postorder(Node* root, vector<int> &v) {
         if( NULL == root) {
             return;
         }
         int i;
         for ( i = 0; i < root->children.size(); i++ ){
               postorder( root->children[i], v);
         }
         v.push_back(root->val);
    } 
};
